Driver/Driver/Util/RepeatTimerManager.swift

```diff
@@ -0,0 +1,91 @@
+//
+//  RepeatTimerManager.swift
+//  Driver
+//
+//  Created by Fatih Doğan on 28.05.2023.
+//
+
+import Foundation
+
+protocol RepeatTimerManagerDelegate: AnyObject {
+    func timeIsUp()
+}
+
+/// This is a class to manage Timer
+/// Use singleton pattern to reach active timer whenever want
+/// For understand to working logic please examine flow charts
+class RepeatTimerManager: NSObject {
+    enum TimerCondition {
+        case resume
+        case stop
+    }
+    // MARK: - Properties
+    private var timer: Timer?
+    private var timeInterval: Double = 15
+    private var timerCondition: TimerCondition = .stop
+    weak var delegate: RepeatTimerManagerDelegate?
+    static let shared = RepeatTimerManager()
+    // MARK: - Init
+    private override init() { }
```

Comment:
It is just a suggestion that you can get this delegate in the initializer as the parameter.

Reply:
In fact I thought to add KVO instead of delegate, but I want to take your opinion too. Where will we implement this manager, in baseviewcontroller or splashscreen etc.? Other question are will we need it more than one instance and How many classes will be communicated with the delegate?

Reply:
What I said with an example 
```suggestion
    private override init(delegate: RepeatTimerManagerDelegate) {
    self.delegate = delegate
     }
```